lib/utils/manage_updates.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:io';
+
+import 'package:sidekick/utils/open_link.dart';
+import 'package:archive/archive.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:github/github.dart';
+//import 'package:hive/hive.dart';
+import 'package:http/http.dart' as http;
+import 'package:oktoast/oktoast.dart';
+import 'package:path_provider/path_provider.dart' as provider;
+import 'package:version/version.dart';
+
+import '../components/molecules/update_card.dart';
+import '../version.dart';
+
+void downloadRelease(String release) async {
+  var directory = await provider.getDownloadsDirectory();
+  var fileLocation =
+      "${directory.absolute.path}${Platform.pathSeparator}CRel-$release.";
+  var url =
+      "https://github.com/aguilaair/Companion/releases/download/$release/${Platform.operatingSystem.toLowerCase()}-$release.zip";
+  var file = File("${fileLocation}zip");
+
+  if (!await file.exists()) {
+    showToast("Downloading...", duration: const Duration(seconds: 30));
+    var res = await http.get(url);
+    await file.writeAsBytes(res.bodyBytes);
+    showToast("Release downloaded! Opening...", dismissOtherToast: true);
+  } else {
+    showToast("File already downloaded, opening...");
+  }
+  openInstaller(file, fileLocation);
+}
+
+void openInstaller(File file, String fileLocation) {
+  if (Platform.isWindows) {
+    installWindows(file, fileLocation);
+  } else if (Platform.isMacOS) {
+    installMacOS(file, fileLocation);
+  } else {
+    openLink("file://${fileLocation.replaceAll("\\", "/")}zip");
+  }
+}
+
+void installWindows(File file, String path) async {
+  var decompressed = ZipDecoder().decodeBytes(file.readAsBytesSync());
+  var msix =
+      decompressed.files.firstWhere((element) => element.name.contains("msix"));
+  await File("${path}msix").writeAsBytes(msix.content);
+  openLink("file://${path.replaceAll("\\", "/")}msix");
+}
+
+void installMacOS(File file, String path) async {
```

Comment:
I think we should download just the DMG if its possible for mac, so we would not need to decode the zip